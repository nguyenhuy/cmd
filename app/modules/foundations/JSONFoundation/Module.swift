Target.module(
  name: "JSONFoundation",
  dependencies: [],
  testDependencies: [])
