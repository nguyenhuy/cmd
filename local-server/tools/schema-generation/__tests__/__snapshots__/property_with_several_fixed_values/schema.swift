// swiftformat:disable all
// This file is generated from ./local-server/src/server/schemas/schemaFile.ts by `yarn export-schema-swift`.
// Do not edit it manually.

import Foundation
import JSONFoundation

extension Schema {
  public struct Message: Codable, Sendable {
    public let role: Role
    public let singleValue = "single_value"
  
    private enum CodingKeys: String, CodingKey {
      case role = "role"
      case singleValue = "single_value"
    }
  
    public init(
        role: Role,
        singleValue: String = "single_value"
    ) {
      self.role = role
    }
  
    public init(from decoder: Decoder) throws {
      let container = try decoder.container(keyedBy: CodingKeys.self)
      role = try container.decode(Role.self, forKey: .role)
    }
  
    public func encode(to encoder: Encoder) throws {
      var container = encoder.container(keyedBy: CodingKeys.self)
      try container.encode(role, forKey: .role)
      try container.encode(singleValue, forKey: .singleValue)
    }
  
    public enum Role: String, Codable, Sendable {
      case system = "system"
      case user = "user"
      case assistant = "assistant"
      case functionCall = "function_call"
    }
  }}
