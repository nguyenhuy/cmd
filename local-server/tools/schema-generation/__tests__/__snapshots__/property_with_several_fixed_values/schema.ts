export interface Message {
  role: "system" | "user" | "assistant" | "function_call";
  single_value: "single_value";
}
