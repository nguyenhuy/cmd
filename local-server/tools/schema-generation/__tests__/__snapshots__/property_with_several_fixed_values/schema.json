{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$ref": "#/definitions/Message",
  "definitions": {
    "Message": {
      "type": "object",
      "properties": {
        "role": {
          "type": "string",
          "enum": [
            "system",
            "user",
            "assistant",
            "function_call"
          ]
        },
        "single_value": {
          "type": "string",
          "const": "single_value"
        }
      },
      "required": [
        "role",
        "single_value"
      ],
      "additionalProperties": false
    }
  }
}